src/network/networkd-sysctl.c

```diff
@@ -40,10 +42,82 @@ static void manager_set_ip_forwarding(Manager *manager, int family) {
                                   enable_disable(t), af_to_ipv4_ipv6(family));
 }
 
+static void sysctl_overwrite_check(sd_event_source *source, const struct inotify_event *event, Hashmap *sysctls) {
+        _cleanup_free_ char *buf = NULL, *path = NULL;
+        const char *dir_path, *value;
+        size_t len;
+        int r;
+
+        assert(event);
+
+        r = sd_event_source_get_inotify_path(source, &dir_path);
+        if (r < 0)
+                return;
+
+        path = path_join(dir_path, event->name);
+        if (!path)
+                return;
+
+        value = hashmap_get(sysctls, path);
+
+        /* We don't handle this sysctl */
+        if (!value)
+                return;
+
+        if (read_full_file(path, &buf, &len) < 0)
+                return;
```

Comment:
so for unexpected failures we really prefer if we'd log at the very least at debug level, it jsut makes things so much easier to debug. i.e. just do:

```
r = read_full_file();
if (r < 0)
        return log_debug_errno(r, …
``

Reply:
if you don't care about the size you read pass NULL as 3rd argument. If you do read_full_file() will automatically do an extra safety check and verify that there's no embedded NUL byte, which is good here, because below you assume what you read is a valid C string

Reply:
actually, for kernel api vfs reads it's safer to use read_virtual_file(), because they have slightly weird semantics.